### PR TITLE
WIP: (fix) make image loading work better

### PR DESCRIPTION
**Pull request by jmars**
Still work in progress, but so far:

* Lazy loads high-res images in the background using interaction manager
* Cancels high-res image loads if you scroll past the image in a flatlist
* Typescript
* Use react-native image components native properties for placeholder/lowres/highres transitions
* Use standard URL parsing for crops and offline/online instead of our own stuff

### CODE REVIEW

File: packages/ad/package.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
     "react": "16.9.0",
     "react-dom": "16.9.0",
     "react-native": "0.59.10",
-    "react-test-renderer": "16.6.3",
+    "react-test-renderer": "16.9.0",
```
Comment (parkjh) [inline packages/ad/package.json]: this upgrade can be a separate pr... @jmars

Comment (jmars) [reply packages/ad/package.json]: Yes although the tests won't pass with the older version, but can rebase once the other PR is in

Comment (parkjh) [reply packages/ad/package.json]: don't wait up for that...

**Comment by jmars** [comment]
Not happy with how this turned out

**Comment by parkjh** [comment]
> Not happy with how this turned out

what happened?
